src/udev/hidraw_id/hidraw_id_desc.h

```diff
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+struct HidrawIdDescResult {
+        bool is_security_token: 1;
+        bool is_u2f_token: 1;
+};
```

Comment:
I'm sorry, I don't fully understand what you would want me to change here. Since something can be both a security token (general term) and a U2F token (a specific instance of a security token, although the only existing one so far...), I am not sure this struct can be cleanly mapped onto an enum. I could turn these flags into int-valued enums and or them together if this is the preferred style in systemd.

Reply:
Not sure, but I thought something the following:
```
typedef enum HidrawIdTokenType {
    HIDRAW_ID_SECURITY_TOKEN = 1 << 0,
    HIDRAW_ID_U2F_TOKEN      = 1 << 1,
} HidrawIdTokenType;
```
and then for example,
```
*ret = HIDRAW_ID_SECURITY_TOKEN | HIDRAW_ID_U2F_TOKEN;
```

(btw, how about to use `typedef` to drop `struct`.)

Reply:
hmm, so the current code only sets both bools or neither, never one without the other. So why have this at all?

Reply:
The reason for that is that `ID_U2F_TOKEN` is set by the original `u2f_hidraw_policy` created by @amluto and for example packaged into Fedora. Since there may already be userspace software that depends on `ID_U2F_TOKEN` instead of the more general `ID_SECURITY_TOKEN`, I decided to keep this in for the sake of backwards compatibility. Would you suggest to remove it?

Reply:
@yuwata I generally prefer structs over enum flags since they are typesafe and easier to query, but can always adapt if you prefer it the other way. I will definitely use `typedef`, thanks for the suggestion.

Reply:
hmm, so these aren#t orthogonal bits, no?

I mean, the current code only sets both or neither, so this doesn't have to be a bitfield or struct at all, but a plain boolean. But even if this code could detect two kinds of tokens: U2F and plain ones, then it every U2F one is a also a security token, right, hence this should just be a plain enum in that case, not a bitfield.

But what am I missing?

Reply:
Yes, the current struct members are not orthogonal, one always implies the other. I originally thought that the struct would make it easier for the caller to take action since they wouldn't have to know about these implications, but agree that this may just be complicating things. I will convert to an enum and we can see whether this is satisfactory.

Reply:
> The reason for that is that `ID_U2F_TOKEN` is set by the original `u2f_hidraw_policy` created by @amluto and for example packaged into Fedora. Since there may already be userspace software that depends on `ID_U2F_TOKEN` instead of the more general `ID_SECURITY_TOKEN`, I decided to keep this in for the sake of backwards compatibility. Would you suggest to remove it?

As far as I know, ID_U2F_TOKEN is only used by u2f_hidraw_policy's scripts -- no one has ever told me they use it.  I plan to deprecate ID_U2F_POLICY once this new code lands in systemd and makes it into a release.